### DISCUSSION TITLE
genfsimg - building an EFI ISO

### DISCUSSION THREAD

**Discussion by sm1000** [ipxe/ipxe#219] [Q&A] [answered]
I am not able to generate EFI ISO. I can provide the configuration files in case I am the only one who sees this error.

distro: **Debian 10.7**
kernel: **Linux 4.19.0-13-amd64**
command: **make bin-i386-efi/ipxe.iso**

Result:

  [DEPS] image/embedded.c
  [BUILD] bin-i386-efi/embedded.o
  [AR] bin-i386-efi/blib.a
ar: creating bin-i386-efi/blib.a
  [VERSION] bin-i386-efi/version.ipxe.efi.o
  [LD] bin-i386-efi/ipxe.efi.tmp
  [FINISH] bin-i386-efi/ipxe.efi
  [GENFSIMG] bin-i386-efi/ipxe.iso
genisoimage: option '-e' is ambiguous; possibilities: '-eltorito-boot' '-eltorito-alt-boot' '-eltorito-catalog' '-exclude' '-exclude-list' '-ethershare' '-exchange'
Usage: genisoimage [options] -o file directory ...

Use genisoimage -help
to get a list of valid options.

Report problems to debburn-devel@lists.alioth.debian.org.
make: *** [Makefile.efi:54: bin-i386-efi/ipxe.iso] Error 1
rm bin-i386-efi/version.ipxe.efi.o

**Comment by mcb30** [comment, marked as answer]
The problem is that the upstream `genisoimage` doesn't support creating EFI boot images.  It seems as though the versions distributed in both Fedora and Ubuntu include patches to support the `-e` option, but the version in Debian does not.

I have pushed commit https://github.com/ipxe/ipxe/commit/7c3d186a1 to check that the required options are supported.  This should allow you to build UEFI ISOs after updating to the current iPXE master branch and installing the `xorriso` package.  (You do not need to uninstall `genisoimage`.)

**Reply by amphineko** [reply]
I'm not able to `make bin-x86_64-efi/ipxe.iso`. It yields `No rule to make target 'bin-i386-efi/ipxe.iso'`.

Is UEFI ISO support still on the way? I'm on the tag `v1.21.1`.

**Reply by mcb30** [reply]
> I'm not able to `make bin-x86_64-efi/ipxe.iso`. It yields `No rule to make target 'bin-i386-efi/ipxe.iso'`.
> 
> Is UEFI ISO support still on the way? I'm on the tag `v1.21.1`.

Use the current master branch instead.

**Reply by amphineko** [reply]
> > I'm not able to `make bin-x86_64-efi/ipxe.iso`. It yields `No rule to make target 'bin-i386-efi/ipxe.iso'`.
> 
> > 
> 
> > Is UEFI ISO support still on the way? I'm on the tag `v1.21.1`.
> 
> 
> 
> Use the current master branch instead.

Should the master branch considered stable enough?

**Reply by mcb30** [reply]
> Should the master branch considered stable enough?

Yes: it's a rolling release model and every commit for the past decade or so has been intended as a production-ready release.

**Reply by amphineko** [reply]
> > Should the master branch considered stable enough?
> 
> 
> 
> Yes: it's a rolling release model and every commit for the past decade or so has been intended as a production-ready release.

Great! Thanks for your help and explanation.
